cpu/stm32f1/periph/gpio.c

```diff
@@ -30,1015 +30,190 @@
+    }
+    else if (pin_num < 10) {
+        NVIC_EnableIRQ(EXTI9_5_IRQn);
     }
-
-    /* set callback */
-    config[gpio_irq].cb = cb;
-    config[gpio_irq].arg = arg;
-
-    /* configure the event that triggers an interrupt */
-    switch (flank) {
-        case GPIO_RISING:
-            EXTI->RTSR |= (1 << exti_line);
-            EXTI->FTSR &= ~(1 << exti_line);
-            break;
-        case GPIO_FALLING:
-            EXTI->RTSR &= ~(1 << exti_line);
-            EXTI->FTSR |= (1 << exti_line);
-            break;
-        case GPIO_BOTH:
-            EXTI->RTSR |= (1 << exti_line);
-            EXTI->FTSR |= (1 << exti_line);
-            break;
+    else {
+        NVIC_EnableIRQ(EXTI15_10_IRQn);
     }
-
     /* clear event mask */
-    EXTI->EMR &= ~(1 << exti_line);
+    EXTI->EMR &= ~(1 << pin_num);
     /* unmask the pins interrupt channel */
-    EXTI->IMR |= (1 << exti_line);
-
+    EXTI->IMR |= (1 << pin_num);
     return 0;
 }
 
-void gpio_irq_enable(gpio_t dev)
+void gpio_init_af(gpio_t pin, gpio_af_t af)
 {
-    uint8_t exti_line;
-
-    switch(dev) {
-#if GPIO_0_EN
-        case GPIO_0:
-            exti_line = GPIO_0_PIN;
-            break;
-#endif
-#if GPIO_1_EN
-        case GPIO_1:
-            exti_line = GPIO_1_PIN;
-            break;
-#endif
-#if GPIO_2_EN
-        case GPIO_2:
-            exti_line = GPIO_2_PIN;
-            break;
-#endif
-#if GPIO_3_EN
-        case GPIO_3:
-            exti_line = GPIO_3_PIN;
-            break;
-#endif
-#if GPIO_4_EN
-        case GPIO_4:
-            exti_line = GPIO_4_PIN;
-            break;
-#endif
-#if GPIO_5_EN
-        case GPIO_5:
-            exti_line = GPIO_5_PIN;
-            break;
-#endif
-#if GPIO_6_EN
-        case GPIO_6:
-            exti_line = GPIO_6_PIN;
-            break;
-#endif
-#if GPIO_7_EN
-        case GPIO_7:
-            exti_line = GPIO_7_PIN;
-            break;
-#endif
-#if GPIO_8_EN
-        case GPIO_8:
-            exti_line = GPIO_8_PIN;
-            break;
-#endif
-#if GPIO_9_EN
-        case GPIO_9:
-            exti_line = GPIO_9_PIN;
-            break;
-#endif
-#if GPIO_10_EN
-        case GPIO_10:
-            exti_line = GPIO_10_PIN;
-            break;
-#endif
-#if GPIO_11_EN
-        case GPIO_11:
-            exti_line = GPIO_11_PIN;
-            break;
-#endif
-#if GPIO_12_EN
-        case GPIO_12:
-            exti_line = GPIO_12_PIN;
-            break;
-#endif
-#if GPIO_13_EN
-        case GPIO_13:
-            exti_line = GPIO_13_PIN;
-            break;
-#endif
-#if GPIO_14_EN
-        case GPIO_14:
-            exti_line = GPIO_14_PIN;
-            break;
-#endif
-#if GPIO_15_EN
-        case GPIO_15:
-            exti_line = GPIO_15_PIN;
-            break;
-#endif
-        default:
-            return;
-    }
-
-    /* unmask the pins interrupt channel */
-    EXTI->IMR |= (1 << exti_line);
-
-    return;
+    (void)pin;
+    (void)af;
+    /* TODO: implement and use this in other peripheral drivers */
```

Comment:
Is this dependent for other peripheral drivers or didn't you want to implement this function right now :-) ?